pkg/cmd/codespace/ssh.go

```diff
@@ -742,58 +341,81 @@ func newCpCmd(app *App) *cobra.Command {
+			ctx, cancel := context.WithCancel(cmd.Context())
+			defer cancel()
+
+			return app.Copy(ctx, args, opts)
 		},
 		DisableFlagsInUseLine: true,
 	}
 
 	// We don't expose all sshOptions.
-	cpCmd.Flags().BoolVarP(&opts.recursive, "recursive", "r", false, "Recursively copy directories")
-	cpCmd.Flags().BoolVarP(&opts.expand, "expand", "e", false, "Expand remote file names on remote shell")
-	opts.selector = AddCodespaceSelector(cpCmd, app.apiClient)
-	cpCmd.Flags().StringVarP(&opts.profile, "profile", "p", "", "Name of the SSH profile to use")
-	return cpCmd
+	cmd.Flags().BoolVarP(&opts.recursive, "recursive", "r", false, "Recursively copy directories")
+	cmd.Flags().BoolVarP(&opts.expand, "expand", "e", false, "Expand remote file names on remote shell")
+	opts.selector = AddCodespaceSelector(cmd, app.apiClient)
+	cmd.Flags().StringVarP(&opts.profile, "profile", "p", "", "Name of the SSH profile to use")
+	return cmd
 }
 
 // Copy copies files between the local and remote file systems.
 // The mechanics are similar to 'ssh' but using 'scp'.
 func (a *App) Copy(ctx context.Context, args []string, opts cpOptions) error {
-	if len(args) < 2 {
-		return fmt.Errorf("cp requires source and destination arguments")
-	}
-	if opts.recursive {
-		opts.scpArgs = append(opts.scpArgs, "-r")
+	tunnel, args, err := a.connect(ctx, opts.selector, args, opts.profile)
+	if err != nil {
+		return err
 	}
+	defer tunnel.Close()
 
-	hasRemote := false
-	for _, arg := range args {
-		if rest := strings.TrimPrefix(arg, "remote:"); rest != arg {
-			hasRemote = true
-			// scp treats each filename argument as a shell expression,
-			// subjecting it to expansion of environment variables, braces,
-			// tilde, backticks, globs and so on. Because these present a
-			// security risk (see https://lwn.net/Articles/835962/), we
-			// disable them by shell-escaping the argument unless the user
-			// provided the -e flag.
-			if !opts.expand {
-				arg = `remote:'` + strings.Replace(rest, `'`, `'\''`, -1) + `'`
-			}
-
-		} else if !filepath.IsAbs(arg) {
-			// scp treats a colon in the first path segment as a host identifier.
-			// Escape it by prepending "./".
-			// TODO(adonovan): test on Windows, including with a c:\\foo path.
-			const sep = string(os.PathSeparator)
-			first := strings.Split(filepath.ToSlash(arg), sep)[0]
-			if strings.Contains(first, ":") {
-				arg = "." + sep + arg
-			}
-		}
-		opts.scpArgs = append(opts.scpArgs, arg)
-	}
-	if !hasRemote {
-		return cmdutil.FlagErrorf("at least one argument must have a 'remote:' prefix")
+	// The cs cp command doesn't have an option to select the server port, we use
+	// 0 here so that it pick an unused port ransomly.
+	if err := tunnel.Forward(ctx, 0); err != nil {
+		return fmt.Errorf("error forwarding tunnel: %w", err)
 	}
-	return a.SSH(ctx, nil, opts.sshOptions)
+
+	return tunnel.Run(opts.profile, func(destination string, port int) error {
+		return codespaces.Copy(ctx, args, port, destination)
+	})
```

Comment:
As the code for connections is encapsulated now, `gh cs cp` doesn't need to depend on  `gh cs ssh`, which was unnecessarily coupling both commands.